test/metrics/index.node.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+/* eslint-env mocha */
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-as-promised'))
+const { expect } = chai
+const sinon = require('sinon')
+
+const { randomBytes } = require('libp2p-crypto')
+const pipe = require('it-pipe')
+const concat = require('it-concat')
+const delay = require('delay')
+
+const { createPeer } = require('../utils/creators/peer')
+const Libp2p = require('../../src')
+const baseOptions = require('../utils/base-options')
+
+describe('libp2p.metrics', () => {
+  let libp2p
+
+  afterEach(async () => {
+    libp2p && await libp2p.stop()
+  })
+
+  it('should disable metrics by default', async () => {
+    [libp2p] = await createPeer({
+      config: {
+        modules: baseOptions.modules
+      }
+    })
+
+    expect(libp2p.metrics).to.not.exist()
+  })
+
+  it('should start/stop metrics on startup/shutdown when enabled', async () => {
+    const config = { ...baseOptions }
+    config.metrics = {
+      enabled: true
+    }
+    libp2p = await Libp2p.create(config)
```

Comment:
use `createPeer` utility to avoid creating a `PeerId`

Reply:
👍 I just noticed the `started` option, I will switch back to using that. I didn't see that option when I was using it before and needed the nodes returned stopped.